src/main/java/org/skriptlang/skript/bukkit/toolcomponent/elements/ExprToolRuleBlocks.java

```diff
@@ -0,0 +1,97 @@
+package org.skriptlang.skript.bukkit.toolcomponent.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.components.ToolComponent.ToolRule;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.toolcomponent.ToolRuleWrapper;
+import org.skriptlang.skript.bukkit.toolcomponent.elements.EffSecCreateToolRule.ToolRuleEvent;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Name("Tool Rule - Blocks")
+@Description("The blocks of a tool rule.")
+@Examples({
+	"create a new tool rule and store it in {_toolrule}:",
+		"\tset the tool rule blocks to oak log, stone and obsidian",
+		"\tset the tool rule speed to 5",
+		"\tenable tool rule drops",
+		"add {_toolrule} to the tool rules of {_item}"
+})
+@RequiredPlugins("Minecraft 1.20.6+")
+@Since("INSERT VERSION")
+public class ExprToolRuleBlocks extends PropertyExpression<ToolRule, ItemType> {
+
+	static {
+		Skript.registerExpression(ExprToolRuleBlocks.class, ItemType.class, ExpressionType.PROPERTY,
+			"[the] tool rule blocks [of %toolrules%]");
```

Comment:
Can we maybe use `block types` over `blocks` prevents an expectation of blocks this will also work better in the future if minecraft ever decides to expand this from just material ids and into blockdata similar to the `can place on` and `can destroy` components

Just fine to say no due to that atm not seemingly being planned